integration/container/devices_windows_test.go

```diff
@@ -0,0 +1,158 @@
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+		{
+			"process/class://5B45201D-F2F2-4F3B-85BB-30FF1F953599 mounted",
+			[]string{"class://5B45201D-F2F2-4F3B-85BB-30FF1F953599"},
+			containertypes.IsolationProcess,
+			false,
+			"",
+			0,
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+		{
+			"process/vpci-class-guid://5B45201D-F2F2-4F3B-85BB-30FF1F953599 mounted",
+			[]string{"vpci-class-guid://5B45201D-F2F2-4F3B-85BB-30FF1F953599"},
+			containertypes.IsolationProcess,
+			false,
+			"",
+			0,
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+		{
+			"hyperv/no device mounted",
+			[]string{},
+			containertypes.IsolationHyperV,
+			false,
+			"",
+			1,
+			"",
+			"",
+		},
+		{
+			"hyperv/class/5B45201D-F2F2-4F3B-85BB-30FF1F953599 mounted",
+			[]string{"class/5B45201D-F2F2-4F3B-85BB-30FF1F953599"},
+			containertypes.IsolationHyperV,
+			!testEnv.RuntimeIsWindowsContainerd(),
+			"device assignment is not supported for HyperV containers",
+			0,
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+		{
+			"hyperv/class://5B45201D-F2F2-4F3B-85BB-30FF1F953599 mounted",
+			[]string{"class://5B45201D-F2F2-4F3B-85BB-30FF1F953599"},
+			containertypes.IsolationHyperV,
+			!testEnv.RuntimeIsWindowsContainerd(),
+			"device assignment is not supported for HyperV containers",
+			0,
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+		{
+			"hyperv/vpci-class-guid://5B45201D-F2F2-4F3B-85BB-30FF1F953599 mounted",
+			[]string{"vpci-class-guid://5B45201D-F2F2-4F3B-85BB-30FF1F953599"},
+			containertypes.IsolationHyperV,
+			!testEnv.RuntimeIsWindowsContainerd(),
+			"device assignment is not supported for HyperV containers",
+			0,
+			"/Windows/System32/HostDriverStore/FileRepository",
+			"",
+		},
+	}
+
+	for _, d := range testData {
+		d := d
+		t.Run(d.doc, func(t *testing.T) {
+			t.Parallel()
+			deviceOptions := []func(*container.TestContainerConfig){container.WithIsolation(d.isolation)}
+			for _, deviceName := range d.devices {
+				deviceOptions = append(deviceOptions, container.WithWindowsDevice(deviceName))
+			}
+
+			id := container.Create(ctx, t, client, deviceOptions...)
+
+			// Hyper-V isolation is failing even with no actual devices added.
+			// !c8d: hcs::CreateComputeSystem <CID>: The request is not supported.
+			// c8d: Error response from daemon: failed to create shim: error in preparing config doc: failed to get host processor information: failed to retrieve processor and processor topology information: hcs::GetServiceProperties: Catastrophic failure: unknown
+			// We have HyperV isolation tests in integration-cli, but all of them skip if the daemon is not defaulting to HyperV isolation.
+			// TODO: Look harder at this, it worked on RS5. Is it just flakey?
+			skip.If(t, d.isolation == containertypes.IsolationHyperV && !d.expectedStartFailure, "FIXME. HyperV isolation setup is probably incorrect in the test")
```

Comment:
As far as I can tell, we currently lack a general "does it work at all" integration test for HyperV-isolation on CI, where process isolation is the default, i.e. Windows Server.

I suspect the above-noted failures are caused by at least some of the test VMs not being configured for nested virtualisation, which is required for VM hosts to run HyperV-isolated containers, but not process-isolated containers.

As such, that probably needs to be investigated elsewhere (and remediated in the CI host setup if I'm correct about the root cause) before this `skip.If` can be removed.

I did mention above that there are HyperV tests, but they are actually tests that are only run when the daemon's default is HyperV, they're not testing HyperV specifically, i.e. [`TestRunHypervIsolationWithCPUCountCPUSharesAndCPUPercent`](https://github.com/moby/moby/blob/master/integration-cli/docker_cli_run_test.go#L4118-L4131).

Reply:
I logged this as #43395 so it can be looked at later.

Reply:
https://github.com/moby/moby/pull/43368/files#r835837598